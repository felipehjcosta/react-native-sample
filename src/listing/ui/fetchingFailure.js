import { Text, View, TouchableHighlight } from 'react-native'
import React from 'react'
import PropTypes from 'prop-types'

const FetchingFailure = ({onRetryButtonTouched}) => {
  return (
    <View style={{
      flex: 1,
      flexDirection: 'column',
      justifyContent: 'center',
      alignItems: 'center'
    }}>
      <Text>Error on fetch data</Text>
      <Text>Verify your internet connection and try gain</Text>
      <TouchableHighlight
        onPress={() => onRetryButtonTouched()}
        underlayColor='#dddddd'
        style={{padding: 10, backgroundColor: '#48BBEC'}}>
        <Text>Retry</Text>
      </TouchableHighlight>
    </View>
  )
}

FetchingFailure.propTypes = {
  onRetryButtonTouched: PropTypes.func.isRequired
}

export default FetchingFailure
