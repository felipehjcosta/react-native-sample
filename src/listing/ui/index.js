// @flow
import {
  FlatList,
  Image,
  StyleSheet,
  Text,
  TouchableHighlight,
  ActivityIndicator,
  View
} from 'react-native'
import PropTypes from 'prop-types'
import React from 'react'
import { withNavigation } from 'react-navigation'

export class ListingUI extends React.Component {

  static propTypes = {
    fetchData: PropTypes.func.isRequired,
    detail: PropTypes.func.isRequired,
    onItemSelected: PropTypes.func.isRequired,
    listingState: PropTypes.object.isRequired,
    navigation: PropTypes.object.isRequired
  }

  componentDidMount () {
    const {
      fetchData
    } = this.props

    fetchData()
  }

  renderLoading () {
    return (
      <View style={{flex: 1, flexDirection: 'row', alignItems: 'center', justifyContent: 'center'}}>
        <ActivityIndicator size='large' />
      </View>
    )
  }

  rowPressed (item) {
    this.props.detail(item)
    this.props.onItemSelected()
  }

  renderRow (row) {
    return (
      <TouchableHighlight
        onPress={() => this.rowPressed(row.item)}
        underlayColor='#dddddd'>
        <View>
          <View style={styles.rowContainer}>
            <Image style={styles.thumb} source={{uri: row.item.img_url}} />
            <View style={styles.textContainer}>
              <Text style={styles.price}>{row.item.price_formatted}</Text>
              <Text style={styles.title} numberOfLines={1}>{row.item.title}</Text>
            </View>
          </View>
          <View style={styles.separator} />
        </View>

      </TouchableHighlight>
    )
  }

  renderList (items) {
    return (
      <FlatList
        data={items}
        renderItem={this.renderRow.bind(this)} />
    )
  }

  render () {
    const {
      listingState
    } = this.props

    if (Object.keys(listingState).length === 0) return false

    if (listingState.isLoading) {
      return this.renderLoading()
    } else {
      return this.renderList(listingState.items)
    }
  }
}

const styles = StyleSheet.create({
  thumb: {
    width: 80,
    height: 80,
    marginRight: 10
  },
  textContainer: {
    flex: 1
  },
  separator: {
    height: 1,
    backgroundColor: '#dddddd'
  },
  price: {
    fontSize: 25,
    fontWeight: 'bold',
    color: '#48BBEC'
  },
  title: {
    fontSize: 20,
    color: '#656565'
  },
  rowContainer: {
    flexDirection: 'row',
    padding: 10
  }
})

export default withNavigation(ListingUI)
