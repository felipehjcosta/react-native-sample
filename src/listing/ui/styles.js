import { StyleSheet } from 'react-native'
import Dimensions from 'Dimensions'

const IMAGE_HEIGHT = Dimensions.get('screen').height * 0.30
const SCREEN_WIDTH = Dimensions.get('screen').width
const IMAGE_SHIMMER_WIDTH = SCREEN_WIDTH - 32
const IMAGE_SHIMMER_HEIGHT = IMAGE_HEIGHT + 48

export default StyleSheet.create({
  itemImage: {
    height: IMAGE_HEIGHT,
    marginLeft: 8,
    marginTop: 8,
    marginRight: 8
  },
  contentContainer: {
    marginLeft: 16,
    marginRight: 16,
    marginTop: 8,
    marginBottom: 8
  },
  itemContainer: {
    flex: 1,
    flexDirection: 'column'
  },
  listing: {
    backgroundColor: '#ffffff'
  },
  contentPrice: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#48BBEC'
  },
  contentTitle: {
    fontSize: 20,
    color: '#656565'
  },
  shimmerPlaceHolder: {
    width: IMAGE_SHIMMER_WIDTH,
    height: IMAGE_SHIMMER_HEIGHT,
    marginLeft: 16,
    marginTop: 16,
    marginBottom: 8
  },
  loadingMoreIndicatorContainer: {
    paddingVertical: 16
  }
})
