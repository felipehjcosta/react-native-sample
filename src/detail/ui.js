import React from 'react'
import { ScrollView, Text, View, Image } from 'react-native'
import PropTypes from 'prop-types'
import styles from './styles.js'

class DetailUI extends React.Component {

  static propTypes = {
    detailState: PropTypes.object.isRequired
  }

  static defaultProps = {}

  render () {
    const {
      detailState
    } = this.props

    const property = detailState
    let stats = property.bedroom_number + ' bed ' + property.property_type
    if (property.bathroom_number) {
      stats += ', ' + property.bathroom_number + ' ' +
        (property.bathroom_number > 1
          ? 'bathrooms' : 'bathroom')
    }

    const price = property.price_formatted

    return (
      <ScrollView styles={{backgroundColor: '#ffffff'}}>
        <View style={styles.container}>
          <Image style={styles.image} source={{uri: property.img_url}} />
          <View style={styles.heading}>
            <Text style={styles.price}>{price}</Text>
            <Text style={styles.title}>{property.title}</Text>
            <View style={styles.separator} />
          </View>
          <Text style={styles.description}>{stats}</Text>
          <Text style={styles.description}>{property.summary}</Text>
        </View>
      </ScrollView>
    )
  }
}

export default DetailUI
