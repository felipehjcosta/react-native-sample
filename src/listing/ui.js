// @flow
import { FlatList, Image, Text, TouchableHighlight, View, ActivityIndicator } from 'react-native'
import PropTypes from 'prop-types'
import React from 'react'
import { withNavigation } from 'react-navigation'
import ShimmerPlaceHolder from 'react-native-shimmer-placeholder'
import styles from './styles.js'

export class ListingUI extends React.Component {

  static propTypes = {
    fetchData: PropTypes.func.isRequired,
    onItemSelected: PropTypes.func.isRequired,
    listingState: PropTypes.object.isRequired,
    navigation: PropTypes.object.isRequired
  }

  componentDidMount () {
    const {
      fetchData
    } = this.props

    fetchData()
  }

  render () {
    const {
      listingState
    } = this.props

    if (Object.keys(listingState).length === 0) return false

    return this.renderList(this.createFlatListViewModel(listingState))
  }

  renderList = (flatListViewModel) => <FlatList
    style={styles.listing}
    contentContainerStyle={flatListViewModel.contentContainerStyle()}
    data={flatListViewModel.items}
    keyExtractor={(item, index) => flatListViewModel.keyExtractor(item, index)}
    renderItem={(row) => flatListViewModel.renderItem(row)}
    refreshing={flatListViewModel.refreshing}
    onRefresh={() => flatListViewModel.onRefresh()}
    onEndReached={() => flatListViewModel.loadMore()}
    onEndReachedThreshold={0.5}
    ListFooterComponent={() => flatListViewModel.renderFooter()}
    ListEmptyComponent={() => flatListViewModel.renderEmpty()}
    ItemSeparatorComponent={this.renderSeparator} />

  renderSeparator = () => <View style={styles.separator} />

  createFlatListViewModel = (listingState) => {
    if (listingState.isLoading) {
      return this.createLoadingFlatListViewModel()
    } else if (listingState.isLoadingFailed) {
      return this.createLoadingFailedFlatListViewModel()
    } else if (listingState.isLoadingMore) {
      return this.createLoadingMoreFlatListViewModel(listingState.items)
    } else if (listingState.isLoadingMoreFailed) {
      return this.createLoadingMoreFailedFlatListViewModel(listingState.items)
    } else {
      return this.createItemsFlatListViewModel(listingState.items)
    }
  }

  createLoadingFlatListViewModel = () => {
    return {
      items: new Array(10),
      keyExtractor: (item, index) => `${index}`,
      renderItem: (row) => this.renderLoadingItem(),
      refreshing: false,
      onRefresh: () => {},
      loadMore: () => {},
      renderFooter: () => false,
      contentContainerStyle: () => {}
    }
  }

  renderLoadingItem = () => {
    return (
      <View>
        <View style={styles.rowContainer}>
          <ShimmerPlaceHolder autoRun style={styles.imageShimmerPlaceHolder} />
          <ShimmerPlaceHolder autoRun style={styles.contentShimmerPlaceHolder} />
        </View>
      </View>
    )
  }

  createLoadingFailedFlatListViewModel = () => {
    return {
      items: [],
      keyExtractor: (item, index) => `${index}`,
      renderItem: (row) => this.renderLoadingItem(),
      refreshing: false,
      onRefresh: () => {},
      loadMore: () => {},
      renderFooter: () => false,
      contentContainerStyle: () => {
        return {
          flex: 1,
          flexDirection: 'column',
          justifyContent: 'center',
          alignItems: 'center'
        }
      },
      renderEmpty: () => this.renderLoadingFailure()
    }
  }

  renderLoadingFailure = () => {
    return (
      <View style={{
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center'
      }}>
        <Text>Error on fetch data</Text>
        <Text>Veify your internet connection and try gain</Text>
        <TouchableHighlight
          onPress={() => this.props.fetchData()}
          underlayColor='#dddddd' style={{padding: 10, backgroundColor: '#48BBEC'}}>
          <Text>Retry</Text>
        </TouchableHighlight>
      </View>
    )
  }

  createLoadingMoreFlatListViewModel = (items) => {
    return {
      items,
      keyExtractor: (item, index) => `${index}`,
      renderItem: (row) => this.renderRow(row),
      refreshing: false,
      onRefresh: () => {},
      loadMore: () => {},
      renderFooter: () => this.renderLoadingMoreFooter(),
      contentContainerStyle: () => {}
    }
  }

  renderLoadingMoreFooter = () => {
    return (
      <View style={styles.loadingMoreIndicatorContainer}>
        <ActivityIndicator animating size='large' />
      </View>
    )
  }

  createLoadingMoreFailedFlatListViewModel = (items) => {
    return {
      items,
      keyExtractor: (item, index) => `${index}`,
      renderItem: (row) => this.renderRow(row),
      refreshing: false,
      onRefresh: () => {},
      loadMore: () => {},
      renderFooter: () => this.renderLoadingMoreFailedFooter(),
      contentContainerStyle: () => {}
    }
  }

  renderLoadingMoreFailedFooter = () => {
    return (
      <View style={{
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center'
      }}>
        <Text>Error on fetch data</Text>
        <Text>Veify your internet connection and try gain</Text>
        <TouchableHighlight
          underlayColor='#dddddd' style={{padding: 10, backgroundColor: '#48BBEC'}}>
          <Text>Retry</Text>
        </TouchableHighlight>
      </View>
    )
  }

  createItemsFlatListViewModel = (items) => {
    return {
      items,
      keyExtractor: (item, index) => `${index}`,
      renderItem: (row) => this.renderRow(row),
      refreshing: false,
      onRefresh: () => this.props.updateData(),
      loadMore: () => this.props.loadMoreData(),
      renderFooter: () => false,
      contentContainerStyle: () => {}
    }
  }

  renderRow = (row) => {
    return (
      <TouchableHighlight
        onPress={() => this.rowPressed(row.item)}
        underlayColor='#dddddd'>
        <View>
          <View style={styles.rowContainer}>
            <Image style={styles.thumb} source={{uri: row.item.img_url}} />
            <View style={styles.textContainer}>
              <Text style={styles.price}>{row.item.price_formatted}</Text>
              <Text style={styles.title} numberOfLines={1}>{row.item.title}</Text>
            </View>
          </View>
        </View>
      </TouchableHighlight>
    )
  }

  rowPressed = (item) => this.props.onItemSelected(item.lister_url)
}

export default withNavigation(ListingUI)
