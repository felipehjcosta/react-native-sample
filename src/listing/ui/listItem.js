import { Text, View, Image } from 'react-native'
import React from 'react'
import PropTypes from 'prop-types'
import styles from './styles'

const ListItem = ({item}) => {
  return (
    <View stye={styles.itemContainer}>
      <Image style={styles.itemImage} source={{uri: item.img_url}} />
      <View style={styles.contentContainer}>
        <Text style={styles.contentPrice}>{item.price_formatted}</Text>
        <Text style={styles.contentTitle} numberOfLines={1}>{item.title}</Text>
      </View>
    </View>
  )
}

ListItem.propTypes = {
  item: PropTypes.shape({
    title: PropTypes.string.isRequired,
    price_formatted: PropTypes.string.isRequired,
    img_url: PropTypes.string,
    lister_url: PropTypes.string
  })
}

export default ListItem
