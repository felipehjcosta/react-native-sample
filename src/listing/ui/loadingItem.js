import { View } from 'react-native'
import React from 'react'
import styles from './styles'
import ShimmerPlaceHolder from 'react-native-shimmer-placeholder'

const LoadingItem = () => {
  return (
    <View style={styles.itemContainer}>
      <ShimmerPlaceHolder autoRun style={styles.shimmerPlaceHolder} />
    </View>
  )
}

export default LoadingItem
