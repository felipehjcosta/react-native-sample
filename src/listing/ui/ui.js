// @flow
import {
  ActivityIndicator,
  FlatList,
  Text,
  TouchableHighlight,
  View
} from 'react-native'
import PropTypes from 'prop-types'
import React from 'react'
import { withNavigation } from 'react-navigation'
import styles from './styles.js'
import LoadingItem from './loadingItem'
import ListItem from './listItem'

export class ListingUI extends React.Component {

  static propTypes = {
    fetchData: PropTypes.func.isRequired,
    onItemSelected: PropTypes.func.isRequired,
    listingState: PropTypes.object.isRequired,
    navigation: PropTypes.object.isRequired
  }

  componentDidMount () {
    const {
      fetchData
    } = this.props

    fetchData()
  }

  render () {
    const {
      listingState
    } = this.props

    if (Object.keys(listingState).length === 0) return false

    return this.renderList(this.createFlatListViewModel(listingState))
  }

  renderList = (flatListViewModel) => <FlatList
    style={styles.listing}
    contentContainerStyle={flatListViewModel.contentContainerStyle()}
    data={flatListViewModel.items}
    keyExtractor={(item, index) => flatListViewModel.keyExtractor(item, index)}
    renderItem={(row) => flatListViewModel.renderItem(row)}
    refreshing={flatListViewModel.refreshing}
    onRefresh={() => flatListViewModel.onRefresh()}
    onEndReached={() => flatListViewModel.loadMore()}
    onEndReachedThreshold={0.5}
    ListFooterComponent={() => flatListViewModel.renderFooter()}
    ListEmptyComponent={() => flatListViewModel.renderEmpty()}
    ItemSeparatorComponent={this.renderSeparator} />

  renderSeparator = () => <View style={styles.separator} />

  createFlatListViewModel = (listingState) => {
    if (listingState.isLoading) {
      return this.createLoadingFlatListViewModel()
    } else if (listingState.isLoadingFailed) {
      return this.createLoadingFailedFlatListViewModel()
    } else if (listingState.isLoadingMore) {
      return this.createLoadingMoreFlatListViewModel(listingState.items)
    } else if (listingState.isLoadingMoreFailed) {
      return this.createLoadingMoreFailedFlatListViewModel(listingState.items)
    } else {
      return this.createItemsFlatListViewModel(listingState.items)
    }
  }

  createLoadingFlatListViewModel = () => {
    return {
      items: new Array(10),
      keyExtractor: (item, index) => `${index}`,
      renderItem: (row) => <LoadingItem />,
      refreshing: false,
      onRefresh: () => {},
      loadMore: () => {},
      renderFooter: () => false,
      contentContainerStyle: () => {},
      renderEmpty: () => { return false }
    }
  }

  createLoadingFailedFlatListViewModel = () => {
    return {
      items: [],
      keyExtractor: (item, index) => `${index}`,
      renderItem: (row) => LoadingItem,
      refreshing: false,
      onRefresh: () => {},
      loadMore: () => {},
      renderFooter: () => false,
      contentContainerStyle: () => {
        return {
          flex: 1,
          flexDirection: 'column',
          justifyContent: 'center',
          alignItems: 'center'
        }
      },
      renderEmpty: () => this.renderLoadingFailure()
    }
  }

  renderLoadingFailure = () => {
    return (
      <View style={{
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center'
      }}>
        <Text>Error on fetch data</Text>
        <Text>Veify your internet connection and try gain</Text>
        <TouchableHighlight
          onPress={() => this.props.fetchData()}
          underlayColor='#dddddd'
          style={{padding: 10, backgroundColor: '#48BBEC'}}>
          <Text>Retry</Text>
        </TouchableHighlight>
      </View>
    )
  }

  createLoadingMoreFlatListViewModel = (items) => {
    return {
      items,
      keyExtractor: (item, index) => `${index}`,
      renderItem: (row) => this.renderRow(row),
      refreshing: false,
      onRefresh: () => {},
      loadMore: () => {},
      renderFooter: () => this.renderLoadingMoreFooter(),
      contentContainerStyle: () => {},
      renderEmpty: () => { return false }
    }
  }

  renderLoadingMoreFooter = () => {
    return (
      <View style={styles.loadingMoreIndicatorContainer}>
        <ActivityIndicator animating size='large' />
      </View>
    )
  }

  createLoadingMoreFailedFlatListViewModel = (items) => {
    return {
      items,
      keyExtractor: (item, index) => `${index}`,
      renderItem: (row) => this.renderRow(row),
      refreshing: false,
      onRefresh: () => {},
      loadMore: () => {},
      renderFooter: () => this.renderLoadingMoreFailedFooter(),
      contentContainerStyle: () => {},
      renderEmpty: () => { return false }
    }
  }

  renderLoadingMoreFailedFooter = () => {
    return (
      <View style={{
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center'
      }}>
        <Text>Error on fetch data</Text>
        <Text>Veify your internet connection and try gain</Text>
        <TouchableHighlight
          underlayColor='#dddddd'
          style={{padding: 10, backgroundColor: '#48BBEC'}}>
          <Text>Retry</Text>
        </TouchableHighlight>
      </View>
    )
  }

  createItemsFlatListViewModel = (items) => {
    return {
      items,
      keyExtractor: (item, index) => `${index}`,
      renderItem: (row) => this.renderRow(row),
      refreshing: false,
      onRefresh: () => this.props.updateData(),
      loadMore: () => this.props.loadMoreData(),
      renderFooter: () => false,
      contentContainerStyle: () => {},
      renderEmpty: () => { return false }
    }
  }

  renderRow = (row) => {
    return (
      <TouchableHighlight
        onPress={() => this.rowPressed(row.item)}
        underlayColor='#dddddd'>
        <ListItem item={row.item} />
      </TouchableHighlight>
    )
  }

  rowPressed = (item) => this.props.onItemSelected(item.lister_url)
}

export default withNavigation(ListingUI)
