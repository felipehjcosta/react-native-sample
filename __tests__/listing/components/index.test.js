// @flow
import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import { Listing } from '../../../src/listing/ui/index'

Enzyme.configure({adapter: new Adapter()})

describe('Listing Component', () => {
  it('should render render correctly', () => {
    const props = {
      fetchData: jest.fn(),
      items: [
        {
          img_url: 'https://imgs.nestimg.com/casa_300_m2_108811252799247473.jpg',
          price_formatted: 'R$ 750.000',
          title: 'Campo Grande,Rio de Janeiro,Rio De Janeiro'
        }
      ]
    }

    const wrapper = shallow(<Listing {...props} />)

    const tree = renderer.create(wrapper).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should call the mock fetch data function ', () => {
    const props = {
      fetchData: jest.fn(),
      items: []
    }

    shallow(<Listing {...props} />)

    expect(props.fetchData.mock.calls.length).toBe(1)
  })
})
