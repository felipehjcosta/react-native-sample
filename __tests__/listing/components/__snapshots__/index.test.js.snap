// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing Component should render render correctly 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 1,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "padding": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://imgs.nestimg.com/casa_300_m2_108811252799247473.jpg",
              }
            }
            style={
              Object {
                "height": 80,
                "marginRight": 10,
                "width": 80,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#48BBEC",
                  "fontSize": 25,
                  "fontWeight": "bold",
                }
              }
            >
              R$ 750.000
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "color": "#656565",
                  "fontSize": 20,
                }
              }
            >
              Campo Grande,Rio de Janeiro,Rio De Janeiro
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#dddddd",
              "height": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
